### PR TITLE
ARROW-3204: [R] Enable R package to be made available on CRAN

### PR DESCRIPTION
**Pull request by nealrichardson**
With these changes, we've got 0.14.0 passing the automated checks and we're pending human review. There will probably be further changes requested, which we'll collect here.

### CODE REVIEW

File: r/inst/NOTICE.txt
Comment (nealrichardson) [inline r/inst/NOTICE.txt]: @wesm is this legit to do? See https://github.com/apache/arrow/pull/4908/files#diff-3d1942ed064c42a43c8aef1fc78fe685R16 for why I put this here. Alternatively we could copy the full NOTICE.txt here, and/or do that at package build time, but this seemed less susceptible to bit rot/human error.

FWIW, there are very few (about 4, AFAICT) packages on CRAN that have ASF source header comments, and most also have this problem (referring to a NOTICE file but no NOTICE file is in the package). [Only one](https://github.com/cran/rkafkajars/blob/master/inst/NOTICE) of those packages includes a NOTICE file (in this same location), but the context is a bit different (the package is just a vehicle for delivering the Java .jar). The most similar package to ours, `sparkR` (with source contained inside the spark monorepo), does not contain a NOTICE file. However, CRAN editorial review of new package submissions has increased in recent months though, and responding "but what about `$PACKAGE` already on CRAN with the same issue?" is a nonstarter. I mention all this just to say that there is not much of a model we can follow here.

Comment (wesm) [reply r/inst/NOTICE.txt]: Ideally NOTICE.txt would be included in the package

Comment (nealrichardson) [reply r/inst/NOTICE.txt]: Ok, done in https://github.com/apache/arrow/pull/4908/commits/0cec0f1e22e22ea497b69f89da0b71a29ce94718

**Comment by nealrichardson** [comment]
Current status: barring other feedback, my plan is to check out the 0.14.1 release branch locally, cherry-pick these commits onto it, and build the revised CRAN submission from that. I could just resubmit 0.14.0, but since 0.14.1 has been released already, it probably makes sense to go with that in order to pick up the Parquet timestamp fix in the C++ libraries. 

Releasing 0.14.1 depends on https://github.com/rwinlib/arrow/pull/2 being merged (cc @jeroen). I think https://github.com/autobrew/homebrew-core/pull/1 would be nice too but is not essential. 

Once CRAN has accepted the package, we can merge this patch, but we probably should not merge until then in case there is more feedback.

**Comment by nealrichardson** [comment]
0.14.1 submission was rejected with some (trivial) feedback from the CRAN humans. That was addressed in https://github.com/apache/arrow/pull/4908/commits/41ca81d2fec84fce1c604880c472917227bb9ed2 and 0.14.1 has been resubmitted, waiting on the next human review.

**Comment by nealrichardson** [comment]
Latest resubmission was accepted: https://cran.r-project.org/package=arrow

This is ready to merge now cc @wesm @romainfrancois 

On CRAN

**Comment by jeroen** [comment]
👍 🎉

**Review by wesm — APPROVED** [review]
+1. Congrats everyone!

**Comment by HyukjinKwon** [comment]
This is super nice!

**Pull request closed without merging**
